### ISSUE TITLE
Broken revdeps relating to #2212

### ISSUE THREAD

**Issue by mikemahoney218** [r-spatial/sf#2277]
As #2212 added an attribute to sf objects, some reverse dependencies (presumably those testing that an object is identical to a previously-saved sf object) started throwing errors, which led to the attribute being gated behind an environment variable. I'm opening this issue to track what packages are impacted as well as to (hopefully) track updates to those packages that will prevent non-breaking additions to the structure of sf objects from causing errors.

I checked all reverse dependencies against 92b949ba0f9c64108e5797433927867bb653d72a , then reverted  b978d7064d8e48db5adea9d96b3c2a34ec0c00f3 and checked all reverse dependencies again. Only five packages had new problems due to the attribute change:

## New problems (5)

|package      |version |error  |warning |note | issue | status |
|:------------|:-------|:------|:-------|:----|:----|:----|
|[geodimension](https://github.com/josesamos/geodimension)|1.0.1   |__+1__ |        |1    |https://github.com/josesamos/geodimension/issues/3 | Fixed on CRAN |
|[h3jsr](https://github.com/obrl-soil/h3jsr)|1.3.1   |__+1__ |        |     | https://github.com/obrl-soil/h3jsr/issues/10; https://github.com/obrl-soil/h3jsr/pull/12| Fixed on GitHub |
|[qualmap](https://github.com/chris-prener/qualmap)|0.2.1   |__+1__ |        |     | https://github.com/chris-prener/qualmap/issues/6; https://github.com/chris-prener/qualmap/pull/7| Fixed on CRAN |
|[riverdist](https://github.com/mbtyers/riverdist)|0.16.1  |__+1__ |        |     | https://github.com/mbtyers/riverdist/issues/19| Fixed on CRAN |
|[tidytransit](https://github.com/r-transit/tidytransit)|1.6.0   |__+1__ |        |1    | https://github.com/r-transit/tidytransit/issues/210| Fixed on CRAN |

h3jsr also errors against 92b949ba0f9c64108e5797433927867bb653d72a, but gets a new error when the attribute is present.

**Comment by edzer** [comment]
Great work, very helpful!

**Comment by mikemahoney218** [comment]
As a bit of an update, I rechecked all the packages my automated script failed on -- all work fine with the new attribute.

That means it's only these five packages, as best as I can tell, that have issues with the new attribute. I've opened issues on each seeing if they're willing to update their tests.

**Comment by mikemahoney218** [comment]
Quick update here -- a month later, 3/5 packages have pushed a fix to their tests to CRAN, with 1 package being fixed on GitHub (yesterday, so hopefully a CRAN release in the future). The last package, qualmap, hasn't had any activity yet -- I just emailed the maintainer to see if he'd be open to the PR I posted on that page.

I have _not_ re-run revdeps since my initial investigation, so it's entirely possible that there are new problem packages -- I'm honestly surprised this only caused 5 issues. But we're getting close to the original issues being fixed.
